src/serialization.cpp

```diff
@@ -152,6 +152,18 @@ size_t JsonWriteRichPresenceObj(char* dest,
                     WriteOptionalString(writer, "join", presence->joinSecret);
                     WriteOptionalString(writer, "spectate", presence->spectateSecret);
                 }
+                else 
+                {
+                    if (presence->buttonUrl[0] && presence->buttonLabel[0] ||
+                        presence->buttonUrl[1] && presence->buttonLabel[1]) {
+                        WriteArray buttons(writer, "buttons");
+                        for (int b = 0; b < DISCORD_MAX_BUTTONS; ++b) {
+                            WriteObject button(writer);
+                            WriteOptionalString(writer, "label", presence->buttonLabel[b]);
```

Comment:
This will write empty string for the first button if the first button is null but the second button is present, is that intended?

Reply:
I will fix it, that not intended, thanks for that notice.